experiments/AMIP/moist_mpi_earth_dynamical_sea_ice/slab_ice/slab_init.jl

```diff
@@ -0,0 +1,213 @@
+
+    return Y, space
+end
+
+"""
+    solve_ice!(dT_sfc, T_sfc, (parameters, F_accumulated), t)
+
+slab RHS with an implicit solve ice and explicit (forward Euler) solve for ocean
+
+"""
+function solve_ice!(integ, Δt)
+
+    Y = integ.u
+    Ya = integ.p.Ya
+    p = integ.p.Ya.params
+
+    # prognostic
+    T_ml = Y.T_ml
+    T_sfc = Y.T_sfc
+    h_ice = Y.h_ice
+
+    # auxiliary
+    F_atm = @. Ya.F_aero + Ya.F_rad
+    ∂F_atm∂T_sfc = get_∂F_atm∂T_sfc(p, T_sfc, Ya) # this will be passed from atmos/SF.jl
+    F_base = similar(T_sfc)
+
+    sm = semster_zero_layer_model.(T_ml, T_sfc, h_ice, F_atm, ∂F_atm∂T_sfc, Ref(p), Ref(Δt))
+
+    T_ml .= sm.T_ml
+    h_ice .= sm.h_ice
+    T_sfc .= sm.T_sfc
+    F_base .= sm.F_base
+
+    # update state
+    @. Y.T_ml = T_ml
+    @. Y.h_ice = h_ice
+    @. Y.T_sfc = T_sfc
+
+    Ya.ice_mask .= get_ice_mask.(h_ice)
+    Ya.F_base .= F_base
+
+    return nothing
+end
+
+
+function semster_zero_layer_model(T_ml, T_sfc, h_ice, F_atm, ∂F_atm∂T_sfc, p, Δt)
+
+    # local
+    ocean_qflux = FT(0)
+    F_base = FT(0)
+
+    # ice thickness and mixed layer temperature changes due to atmosphereic and ocean fluxes 
+    if h_ice > 0 # ice-covered 
+        F_base = p.F0_base * (T_ml - p.T_base)
+        ΔT_ml = -(F_base + ocean_qflux) * Δt / (p.h_ml * p.ρc_ml)
+        Δh_ice = (F_atm - F_base) * Δt / p.L_ice
+    else # ice-free
+        ΔT_ml = -(F_atm + ocean_qflux) * Δt / (p.h_ml * p.ρc_ml)
+        Δh_ice = 0
+    end
+
+    # adjust if transition to ice-covered 
+    if (T_ml[1] + ΔT_ml[1] < p.T_freeze)
+        Δh_ice = Δh_ice - (T_ml + ΔT_ml - p.T_freeze) * (p.h_ml * p.ρc_ml) / p.L_ice
+        ΔT_ml = p.T_freeze - T_ml
+    end
+
+    # adjust if transition to ice-free
+    if ((h_ice[1] > 0) & (h_ice[1] + Δh_ice[1] <= 0))
+        ΔT_ml = ΔT_ml - (h_ice + Δh_ice) * p.L_ice / (p.h_ml * p.ρc_ml)
+        Δh_ice = -h_ice
+    end
+
+    # solve for T_sfc
+    if (h_ice[1] + Δh_ice[1] > 0) #  surface is ice-covered
+        # if ice covered, solve implicity (for now one Newton iteration: ΔT_s = - F(T_s) / dF(T_s)/dT_s )
+        F_conductive = p.k_ice / (h_ice + Δh_ice) * (p.T_base - T_sfc)
+        ΔT_sfc = (-F_atm + F_conductive) / (p.k_ice / (h_ice + Δh_ice) + ∂F_atm∂T_sfc)
+        if (T_sfc[1] + ΔT_sfc[1] > p.T_freeze)
+            ΔT_sfc = p.T_freeze - T_sfc
+        end
+        # surface is ice-covered, so update T_sfc as ice surface temperature
+
+        T_sfc += FT(ΔT_sfc)
+    else # ice-free, so update T_sfc as mixed layer temperature
+        T_sfc = FT(T_ml + ΔT_ml)
+    end
+
+    T_ml += FT(ΔT_ml)
+    h_ice += FT(Δh_ice)
+
+    return (; T_ml = FT(T_ml), h_ice = FT(h_ice), F_base = FT(F_base), T_sfc = FT(T_sfc))
+end
+
+get_∂F_atm∂T_sfc(p, T_sfc, Ya) = @. FT(4) * (FT(1) - p.α) * p.σ * T_sfc^3 + Ya.∂F_aero∂T_sfc
+
+function ∑tendencies_ice_stub(du, u, p, t)
+    dY = du
+    Y = u
+    FT = eltype(dY)
+
+    if p.Ya.prescribed_sic_data !== nothing # Prognostic eqn for T_sfc
```

Comment:
I agree with this plan. I'm tempted to leave it in  for now though until we revamp the interface, as it is useful to have the prescribed_sic option as a check in case the Dynamic one is doing something funky.